src/core/Carousel/Carousel.tsx

```diff
@@ -113,35 +114,64 @@ const CarouselSlider = (props: React.ComponentPropsWithoutRef<'ol'>) => {
     () =>
       React.Children.map(props.children, (child, index) =>
         React.isValidElement(child)
-          ? React.cloneElement(child, { id: `${idPrefix}--slide-${index}` })
+          ? React.cloneElement(child, {
+              id: `${idPrefix}--slide-${index}`,
+              index,
+            })
           : child,
       ) ?? [],
     [props.children, idPrefix],
-  );
+  ) as React.ReactNode[];
+
+  React.useEffect(() => {
+    if (!sliderRef.current) {
+      return;
+    }
+
+    sliderRef.current.children.item(currentIndex)?.scrollIntoView();
```

Comment:
What happens if auto switching is on in carousel and user has scrolled down the page so the carousel is not in the viewport anymore? Will it scroll the page up?

Reply:
Ok, still strange behavior 😕
![carousel-scroll-top](https://user-images.githubusercontent.com/36186912/158585086-aa83bc37-8f0e-4cea-8334-c4f8f3d08b3f.gif)

Reply:
This seems to fix the issue: 
![image](https://user-images.githubusercontent.com/36186912/158585616-76c44bc1-ac22-4b5d-929d-8f44f64188b7.png)

Reply:
Fixed and added Safari fallback
![image](https://user-images.githubusercontent.com/9084735/158882764-761ee22e-cd6a-4e68-9fc7-b1a7c03fd674.png)

Would be good if @gretanausedaite can check

Reply:
Will it throw an error if not supported?

Reply:
> Will it throw an error if not supported?

That's also something @gretanausedaite could check.

Reply:
Oh ell, looks like this will not get tested on macos, so I've removed the try/catch and replaced `scrollIntoView` with _only_ `scrollTo`. It feels somewhat jittery on mobile emulation but works ok otherwise.

Reply:
Confirmed is working as expected in Safari on macOS.

Reply:
![carousel-resize-hide](https://user-images.githubusercontent.com/36186912/160399747-87536a50-9285-40a9-8a5b-064487a1732f.gif)

Reply:
can you stop resizing everything? 🔪 😄

Reply:
Just doing basic QA stuff 😀

Reply:
should be good now [`8bef04f`](https://github.com/iTwin/iTwinUI-react/pull/569/commits/8bef04f229db88e5497f9ab6692002b66afa49ed)